Controller/Callback/Offsite.php

```diff
@@ -66,7 +67,7 @@ public function execute()
             return $this->redirect(self::PATH_CART);
         }
 
-        if ($order->getState() !== Order::STATE_PENDING_PAYMENT) {
+        if (! in_array($order->getState(), [ Order::STATE_PENDING_PAYMENT, Order::STATE_PROCESSING ])) {
```

Comment:
@jacstn Did you test access this callback url with an order that paid with Alipay, Internet Banking, and Installment payment? Wonder what is going to happen.

As this original ` if ($order->getState() !== Order::STATE_PENDING_PAYMENT) { ... }` only allow an order with `pending` status to go through.

Reply:
I tested it.. everything except `Truemone` brings here `pending`, only `truemoney` firstly brings here first `pending` and then second callback `processing`.

Reply:
@jacstn no, I mean, what if we "access" the url directly after being redirected here.
For example, I made a charge with Internet Banking, then the system brings me to the return uri: `https://www.mystore/callback/offsite/omise`. And the process went through as normal.

But somehow, I know this url and I attempt to access this one more time, what would happen in this case, with the order that has been paid with Internet Banking?

Reply:
so do you mean @guzzilar:
1. payment is made
2. status is updated with a callback as `pending`
3. webhook come, and payment is completed
4. again callback comes from point 2 with the same, `pending` or `processing`

if this will happen, the merchant will have the wrong status of payment according to my knowledge, the only good status will be visible in omise.co

Reply:
What about if we just ignoried the second hit on the URL and just deal with just the webhook as normal? Also, have we asked the back end devs why this second hit on the return URL is happening? Is there a reason for it?

Reply:
indeed we do ignore, but still, we need to redirect.